File: src/tf_demo/scripts/random_track.py
from random import uniform

import rospy
from geometry_msgs.msg import Twist


def main():
    rospy.init_node("random_track")
    publisher = rospy.Publisher("/turtle1/cmd_vel", Twist, queue_size=1000)

    twist = Twist()

    rate = rospy.Rate(1)
    while not rospy.is_shutdown():
        twist.linear.x = uniform(-1.3, 1.3)
        twist.angular.z = uniform(-1.3, 1.3)
        publisher.publish(twist)
        rate.sleep()


if __name__ == "__main__":
    main()

File: src/tf_demo/src/random_track.cpp
#include "geometry_msgs/Twist.h"
#include "ros/ros.h"

#include <random>

double generate_random() {
    std::random_device rand;
    std::mt19937 gen(rand());
    std::uniform_real_distribution<> dis(-1.3, 1.3);
    return dis(gen);
}

int main(int argc, char **argv) {
    setlocale(LC_ALL, "");
    ros::init(argc, argv, "random_track");
    ros::NodeHandle n;

    ros::Publisher publisher = n.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel", 1000);
    geometry_msgs::Twist twist;

    ros::Rate rate(1);
    while (ros::ok()) {
        twist.linear.x = generate_random();
        twist.angular.z = generate_random();
        publisher.publish(twist);
        ros::spinOnce();
        rate.sleep();
    }
    return 0;
}